modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/package.scala

```diff
@@ -38,7 +38,7 @@ package object resolver {
    * Json in case of success or Map of all currently failed repositories
    * in case of failure
    */
-  type SchemaLookup = Either[LookupFailureMap, Json]
+  type SchemaLookup = Either[LookupFailureMap, (Json, Option[SchemaVer.Full])]
```

Comment:
I suggest making `(Json, Option[SchemaVer.Full])` a case class. It will make client code more readable and cleaner. As you'd give a meaningful names to the fields and would be able to provide a default constructor for None.